mainProject/controllers/users.ts

```diff
@@ -1,119 +1,76 @@
-import { User } from '../models/user';
-import { isAvailableProperty } from '../utils/sort';
-import { v4 } from 'uuid';
+import { userService } from "../services/userService";
 
-const localStorage: Array<User> = [
-    {
-        id: v4(),
-        login: 'some',
-        password: 'pass',
-        age: 11,
-        isDeleted: false
-    }
-];
-
-const getAutoSuggestUsers = (
-    loginSubstring: string,
-    limit: number
-): Array<User> | [] => {
-    const availableRecords = localStorage
-        .filter((obj) => !obj.isDeleted)
-        .sort((firstObj, secondObj) => {
-            if (firstObj.login > secondObj.login) return 1;
-            if (firstObj.login < secondObj.login) return -1;
-            return 0;
-        });
-    const result = [];
-    for (let i = 0; i < availableRecords.length; i++) {
-        if (result.length === limit) {
-            break;
-        }
-        if (availableRecords[i].login.includes(loginSubstring)) {
-            result.push(availableRecords[i]);
-        }
-    }
-
-    return result;
-};
-
-const getUsers = (req, res) => {
-    const query = req.query;
-    if (
-        !query.limit ||
+const getUsers = async (req, res) => {
+  const query = req.query;
+  if (
+    !query.limit ||
     !Number.isInteger(Number(query.limit)) ||
     Number(query.limit) < 0
-    ) {
-        return res.status(400).send('Property limit should be Integer and not less than 0.');
-    }
-    if (!query.login) {
-        return res.status(400).send('Property login should be in the query and have a value.');
-    }
-    const responseData = getAutoSuggestUsers(
-        String(query.login),
-        Number(query.limit)
-    );
-    res.json(responseData);
+  ) {
+    return res
+      .status(400)
+      .send("Property limit should be Integer and not less than 0.");
+  }
+  if (!query.login) {
+    return res
+      .status(400)
+      .send("Property login should be in the query and have a value.");
+  }
+  const responseData = await userService.getAutoSuggestUsers(
+    String(query.login),
+    Number(query.limit)
+  );
+  res.json(responseData);
 };
 
-const createUser = (req, res) => {
-    const user = req.body;
-    const dataKeys = Object.keys(user);
-    if (!dataKeys.length || isAvailableProperty(dataKeys)) {
-        return res.status(400).send('Incorrect request data.');
-    }
-    const userId = v4();
-    const userWithId: User = {
-        id: userId,
-        ...user,
-        isDeleted: false
-    };
-    localStorage.push(userWithId);
+const createUser = async (req, res) => {
+  const user = req.body;
 
-    res.send(`User with login ${userWithId.login} was added to the Database.`);
+  const data = await userService.createUser(user);
+
+  res.send(`User with login ${data.login} was added to the Database.`);
```

Comment:
I don’t think so, since we already have information about the created user in the request, but I see the point in the response being the id of the created user